canvas_sdk/handlers/simple_api/security.py

```diff
@@ -114,6 +114,39 @@ def __init__(self, request: "Request") -> None:
         self.key = key
 
 
+class SessionCredentials(Credentials):
+    """
+    Session credentials class.
+
+    Looks for headers set by Canvas with information about the logged in user based on browser
+    cookies and session information in the Canvas database. These headers are removed if received by
+    the client and intentionally set by Canvas only if there is a valid session from an active user.
+
+    One must look at the type of user in order to understand what the id refers to – a patient or a
+    staff member.
+
+    If a SimpleAPI handler is protected by SessionCredentials, the non-presence of these headers
+    results in an unauthorized response. Users of SessionCredentials will likely wish to further
+    restrict access based on the type of user (patient or staff) and perhaps based on the roles
+    associated with the user. This logic is the concern of the implementer of the handler's
+    authenticate method.
+    """
+
+    def __init__(self, request: "Request") -> None:
+        super().__init__(request)
+
+        if (
+            "x-canvas-logged-in-user-type" not in request.headers
+            or "x-canvas-logged-in-user-id" not in request.headers
+        ):
+            raise NoAuthorizationHeaderError
+
+        self.logged_in_user = {
+            "id": request.headers.get("x-canvas-logged-in-user-id"),
+            "type": request.headers.get("x-canvas-logged-in-user-type"),
```

Comment:
X- headers are considered deprecated and not best practice: https://datatracker.ietf.org/doc/html/rfc6648

So I'd just go with a `Canvas-` prefix.

```suggestion
            "id": request.headers.get("canvas-logged-in-user-id"),
            "type": request.headers.get("canvas-logged-in-user-type"),
```